File: apps/backend/update-templates.js
const { Pool } = require('pg');
const fs = require('fs');
const path = require('path');

const pool = new Pool({
  host: 'localhost',
  port: 5433,
  database: 'aluplan_dev',
  user: 'postgres',
  password: 'postgres',
});

const templates = [
  {
    name: 'Standard Certificate',
    file: 'standard-template.html',
    description: 'Standard certificate - Clean classic design (optimized for single-page PDF)'
  },
  {
    name: 'Premium Certificate',
    file: 'premium-template.html',
    description: 'Premium certificate - Modern gradient design with badge (optimized for single-page PDF)'
  },
  {
    name: 'Executive Certificate',
    file: 'executive-template.html',
    description: 'Executive certificate - Dark formal design with double borders (optimized for single-page PDF)'
  }
];

async function updateTemplates() {
  const client = await pool.connect();
  
  try {
    console.log('📝 Updating certificate templates...\n');
    
    for (const template of templates) {
      const filePath = path.join(__dirname, 'src/modules/certificates/templates', template.file);
      
      if (!fs.existsSync(filePath)) {
        console.log(`⚠️  File not found: ${filePath}`);
        continue;
      }
      
      const htmlContent = fs.readFileSync(filePath, 'utf-8');
      
      const result = await client.query(`
        UPDATE certificate_templates 
        SET "htmlContent" = $1, 
            description = $2,
            "updatedAt" = NOW()
        WHERE name = $3
        RETURNING id, name
      `, [htmlContent, template.description, template.name]);
      
      if (result.rows.length > 0) {
        console.log(`✅ Updated ${result.rows[0].name} template (ID: ${result.rows[0].id})`);
      } else {
        console.log(`⚠️  Template '${template.name}' not found in database`);
      }
    }
    
    // Show summary
    console.log('\n📋 Template Summary:');
    const summary = await client.query(`
      SELECT id, name, description, LENGTH("htmlContent") as html_length, "updatedAt" 
      FROM certificate_templates 
      ORDER BY name
    `);
    
    summary.rows.forEach(row => {
      console.log(`  - ${row.name}: ${row.description} (${row.html_length} bytes)`);
    });
    
    console.log('\n🎉 Certificate templates updated successfully!');
    console.log('💡 You can now generate certificates and they should be single-page PDFs.');
    console.log('\n📏 Optimizations applied:');
    console.log('  - Larger certificate containers (287mm × 200mm)');
    console.log('  - Reduced internal padding and margins (20-33%)');
    console.log('  - Smaller fonts (10-15% reduction)');
    console.log('  - Enhanced visual distinction between templates');
    console.log('  - Description field support added');
    
  } catch (error) {
    console.error('❌ Error updating templates:', error);
    throw error;
  } finally {
    client.release();
    await pool.end();
  }
}

updateTemplates().catch(console.error);

File: apps/backend/update-templates.py
#!/usr/bin/env python3
import psycopg2
import os
from pathlib import Path

# Database connection
conn = psycopg2.connect(
    dbname="aluplan_dev",
    user="postgres",
    password="postgres",
    host="localhost",
    port="5433"
)

cur = conn.cursor()

# Template files
templates_dir = Path(__file__).parent / "src/modules/certificates/templates"

templates = [
    {
        "name": "Default",
        "file": "standard-template.html",
        "description": "Standard certificate - Clean classic design (optimized for single-page PDF)"
    },
    {
        "name": "Premium",
        "file": "premium-template.html",
        "description": "Premium certificate - Modern gradient design with badge (optimized for single-page PDF)"
    },
    {
        "name": "Executive",
        "file": "executive-template.html",
        "description": "Executive certificate - Dark formal design with double borders (optimized for single-page PDF)"
    }
]

print("📝 Updating certificate templates...")

for template in templates:
    file_path = templates_dir / template["file"]
    
    if not file_path.exists():
        print(f"⚠️  File not found: {file_path}")
        continue
    
    with open(file_path, 'r', encoding='utf-8') as f:
        html_content = f.read()
    
    cur.execute("""
        UPDATE certificate_templates 
        SET html_content = %s, 
            description = %s,
            updated_at = NOW()
        WHERE name = %s
        RETURNING id, name
    """, (html_content, template["description"], template["name"]))
    
    result = cur.fetchone()
    if result:
        print(f"✅ Updated {result[1]} template (ID: {result[0]})")
    else:
        print(f"⚠️  Template '{template['name']}' not found in database")

conn.commit()

# Show updated templates
print("\n📋 Template Summary:")
cur.execute("""
    SELECT id, name, description, LENGTH(html_content) as html_length, updated_at 
    FROM certificate_templates 
    ORDER BY name
""")

for row in cur.fetchall():
    print(f"  - {row[1]}: {row[2]} ({row[3]} bytes)")

cur.close()
conn.close()

print("\n🎉 Certificate templates updated successfully!")
print("💡 You can now generate certificates and they should be single-page PDFs.")
